pkgs/servers/nextcloud/packages/apps/memories.nix

```diff
@@ -0,0 +1,89 @@
+{
+  stdenv,
+  buildGoModule,
+  exiftool,
+  fetchurl,
+  ffmpeg,
+  fetchFromGitHub,
+  lib,
+
+  ncVersion,
+}:
+let
+  latestVersionForNc = {
+    "31" = latestVersionForNc."29";
+    "30" = latestVersionForNc."29";
+    "29" = {
+      version = "7.5.2";
+      appHash = "sha256-BfxJDCGsiRJrZWkNJSQF3rSFm/G3zzQn7C6DCETSzw4=";
+      srcHash = "sha256-imBO/64NW5MiozpufbMRcTI9WCaN8grnHlVo+fsUNlU=";
+    };
+  };
+  currentVersionInfo =
+    latestVersionForNc.${ncVersion}
+      or (throw "memories currently does not support nextcloud version ${ncVersion}");
+
+  commonMeta = with lib; {
+    homepage = "https://apps.nextcloud.com/apps/memories";
+    changelog = "https://github.com/pulsejet/memories/blob/v${currentVersionInfo.version}/CHANGELOG.md";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ SuperSandro2000 ];
+  };
+
+  go-vod = buildGoModule rec {
+    pname = "go-vod";
+    inherit (currentVersionInfo) version;
+
+    src = fetchFromGitHub {
+      owner = "pulsejet";
+      repo = "memories";
+      tag = "v${version}";
+      hash = currentVersionInfo.srcHash;
+    };
+
+    sourceRoot = "${src.name}/go-vod";
+
+    vendorHash = null;
+
+    meta = commonMeta // {
+      description = "Extremely minimal on-demand video transcoding server in go";
+      mainProgram = "go-vod";
+    };
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "nextcloud-app-memories";
+  inherit (currentVersionInfo) version;
+
+  src = fetchurl {
+    url = "https://github.com/pulsejet/memories/releases/download/v${version}/memories.tar.gz";
+    hash = currentVersionInfo.appHash;
+  };
+
+  postPatch = ''
+    ln -s ${lib.getExe go-vod} bin-ext/go-vod
+    for bin in bin-ext/{go-vod-aarch64,go-vod-amd64}; do
```

Comment:
Does memories assume that one of those exists?
If so, I think we'd also have to restrict meta.platforms, right?

Reply:
It looks for the one coincidentally named right, and then it sets it if you do `occ maintenance:repair`
https://github.com/pulsejet/memories/blob/372098fd5e67e044db42f533261539b34293c77b/lib/Service/BinExt.php#L138-L168

Reply:
I'd vouch for patching the entire method tbh.

Reply:
That file often changes and I don't really want to maintain patches. The main goal was to remove the vendored binaries here and replace them with symlinks to our system binaries.